astropy/utils/misc.py

```diff
@@ -815,9 +818,29 @@ def __init__(cls, cls_name, bases, members):
             instances = OrderedDict((key, value) for value, key in instances)
             setattr(cls, descriptor_cls._class_attribute_, instances)
 
-        super().__init__(cls_name, bases, members)
+        super(OrderedDescriptorContainer, cls).__init__(cls_name, bases,
+                                                        members)
 
 
+def get_parameters(members):
+    """
+    Looks for ordered descriptors in a class definition and
+    copies such definitions in two new class attributes,
+    one being a dictionary of these objects keyed by their
+    attribute names, and the other a simple list of those names.
+
+    """
+    pdict = OrderedDict()
+    for name, obj in members.items():
+        if (not isinstance(obj, OrderedDescriptor)):
+            continue
+        if obj._name_attribute_ is not None:
+            setattr(obj, '_name', name)
+        pdict[name] = obj
+
+    members['_parameter_vals_'] = pdict
+    members['_parameters_'] = pdict.keys()
```

Comment:
Why not return only members['_parameter_vals_'] and assign it to modeling._MetaModel._parameters_ . It will simplify code in modeling (if I'm not missing something, which is quite possible).